File: src/main/java/kr/co/bookand/backend/policy/domain/Policy.java
package kr.co.bookand.backend.policy.domain;

import kr.co.bookand.backend.common.domain.BaseEntity;
import lombok.*;

import javax.persistence.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Builder
public class Policy extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "policy_id")
    private Long id;

    private String name;
    private String title;

    @Column(columnDefinition = "TEXT", length = 30000)
    private String content;

    public void updateContent(String content) {
        this.content = content;
    }
}

File: src/main/kotlin/kr/co/bookand/backend/policy/domain/Policy.kt
package kr.co.bookand.backend.policy.domain

import kr.co.bookand.backend.common.domain.BaseEntity
import javax.persistence.*

@Entity
class Policy(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "policy_id")
    var id: Long = 0,
    var name: String,
    var title: String,
    @Column(columnDefinition = "text", length = 30000)
    var content: String
) : BaseEntity() {
    fun updateContent(content: String) {
        this.content = content
    }
}
